android/src/main/java/ee/forgr/audio/NativeAudio.java

```diff
@@ -255,6 +283,276 @@ public void run() {
+
+                        // Track this as a playOnce asset
+                        plugin.playOnceAssets.add(assetId);
+
+                        // Store notification metadata if provided
+                        JSObject metadata = call.getObject(NOTIFICATION_METADATA);
+                        if (metadata != null) {
+                            Map<String, String> metadataMap = new HashMap<>();
+                            if (metadata.has("title")) metadataMap.put("title", metadata.getString("title"));
+                            if (metadata.has("artist")) metadataMap.put("artist", metadata.getString("artist"));
+                            if (metadata.has("album")) metadataMap.put("album", metadata.getString("album"));
+                            if (metadata.has("artworkUrl")) metadataMap.put("artworkUrl", metadata.getString("artworkUrl"));
+                            if (!metadataMap.isEmpty()) {
+                                plugin.notificationMetadataMap.put(assetId, metadataMap);
+                            }
+                        }
+
+                        // Preload the asset using the helper method
+                        try {
+                            // Check if asset already exists
+                            if (plugin.audioAssetList.containsKey(assetId)) {
+                                throw new Exception(ERROR_AUDIO_EXISTS + " - " + assetId);
+                            }
+
+                            // Load the asset using the helper method
+                            JSObject headersObj = call.getObject("headers");
+                            AudioAsset asset = plugin.loadAudioAsset(assetId, assetPath, isLocalUrl, volume, audioChannelNum, headersObj);
+
+                            if (asset == null) {
+                                throw new Exception("Failed to load asset");
+                            }
+
+                            // Add to asset list; completion listener is set below with cleanup
+                            plugin.audioAssetList.put(assetId, asset);
+
+                            // Store the file path if we need to delete it later
+                            final String filePathToDelete;
+                            if (deleteAfterPlay && isLocalUrl && assetPath.startsWith("file://")) {
+                                filePathToDelete = assetPath;
+                            } else {
+                                filePathToDelete = null;
+                            }
+
+                            // Set up completion listener for automatic cleanup
+                            asset.setCompletionListener(
+                                new AudioCompletionListener() {
+                                    /**
+                                     * Handles completion for a play-once asset by dispatching the completion event and performing cleanup.
+                                     *
```

Comment:
Avoid using a single `*` in javadocs. Use `* <p>` instead

```suggestion
                                     * <p>
```

Here is the difference:

<img width="1244" height="920" alt="image" src="https://github.com/user-attachments/assets/3afdc585-6534-46dc-a64d-41a9cd4f065c" />

<img width="1234" height="958" alt="image" src="https://github.com/user-attachments/assets/b177a73c-eead-4d36-ab2c-89cf78962543" />